framework/di/Container.php

```diff
@@ -522,14 +529,28 @@ protected function getDependencies($class)
                     try {
                         $c = $param->getClass();
                     } catch (ReflectionException $e) {
-                        $c = null;
+                        if (!$this->isNulledParam($param)) {
+                            $notInstantiableClass = null;
+                            if (PHP_VERSION_ID >= 70000) {
+                                $type = $param->getType();
+                                if ($type instanceof ReflectionNamedType) {
+                                    $notInstantiableClass = $type->getName();
+                                }
+                            }
+                            throw new NotInstantiableException(
```

Comment:
I think this should be `InvalidConfigException` too, if we want to retain BC.

Reply:
Before this PR on some cases throws `NotInstantiableException` and in others cases throws `InvalidConfigException`.
After this PR always `NotInstantiableException`.

BC not break, because `NotInstantiableException` extends `InvalidConfigException`.

Reply:
Ah, my bad! In that case LGTM!